ci/jenkins/stop-jobs.sh

```diff
@@ -0,0 +1,155 @@
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --pull-request)
+    PULL_REQUEST_ID="$2"
+    shift 2
+    ;;
+    --jenkins)
+    JENKINS_URL="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *) 
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+function abort_queue_jobs {
+    echo "====== checking for capv stale waiting jobs on PR ${PULL_REQUEST_ID} ======"
+    queue="$(curl -s -XGET "${JENKINS_URL}queue/api/json" )"
+    length_of_queue="$(echo "$queue" | jq '[.[]][2]' | jq length)"
+
+    for (( j=0; j<$length_of_queue; j++ )); do
+            # fetch the waiting job id and its pr id.
+            waiting_job_pr_id=$(echo "$queue" | jq "[.[]][2][$j].actions[1].parameters[9].value" | tr -d '"')
+            job_name=$(echo "$queue" | jq "[.[]][2][$j].task.name" | tr -d '"')
+
+            if [[ "$PULL_REQUEST_ID" == "$waiting_job_pr_id" ]]; then
+                echo "====== abort waiting job $job_name on PR ${PULL_REQUEST_ID} ======"
+                job_id=$(echo "$queue" | jq "[.[]][2][$j].id")
+                curl -XPOST "${JENKINS_URL}/queue/cancelItem?id=$job_id" -u admin:$JENKINS_API_TOKEN
+            fi
+    done
+}
+
+function abort_running_jobs {
+    echo "====== checking for capv stale running jobs on PR ${PULL_REQUEST_ID} ======"
+    capv_jobs="$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json")"
+    length_of_capv_jobs=$(echo "$capv_jobs" | jq "[.[]][10]" | jq length)
+
+    for ((i=0; i<$length_of_capv_jobs; i++)); do
+    
+        capv_job_name=$(echo "$capv_jobs" | jq "[.[]][10][$i].name" | tr -d '"')
+
+        # job color is the status of the jobs, for example if jobs are in running state then its value will be {"red_anime","blue_anime","aborted_anime".....}
+        # so here "*_anime" means job is in running state.
+        # red_anime: last failed job is in running state.
+        # blue_anime: last successful job is in running state.
+        # aborted_anime: last aborted job is in running state.   
+        job_color=$(echo "$capv_jobs" | jq "[.[]][10][$i].color" | tr -d '"')
+
+        if [[ ${job_color: -5} == "anime" ]]; then
+            last_build_number=$(curl -s -XGET "${JENKINS_URL}job/$capv_job_name/api/json" | jq "[.[]][10][0].number" | tr -d '"')
+            jenkins_varriable="$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json")"
+            pr_id=$(echo "$jenkins_varriable" | jq "[.[]][1].ghprbPullId"| tr -d '"')
+
+            if [[ "$PULL_REQUEST_ID" == "$pr_id" ]]; then
+                echo "====== abort running job $capv_job_name on PR ${PULL_REQUEST_ID} ======"
+
+                # need to stop first then clean.
+                curl -XPOST "${JENKINS_URL}job/$capv_job_name/$last_build_number/stop" -u admin:$JENKINS_API_TOKEN
```

Comment:
Do we need to check if stop is successful or not?

Reply:
> Do we need to check if stop is successful or not?

I think yes, we can put an 'if' condition here to check whether the job has been stopped or not.